### ISSUE TITLE
Column widths of the joined columns should autoadjust according to joined header text.

### ISSUE THREAD

**Issue by vaadin-bot** [vaadin/framework#7294]
**Originally by @TatuLund**

___
Would it be possible to tell the grid to dynamically size based upon joined columns contents OR the joined header’s text? For example, it seems to work well with a single column without any extra headers.

Longer title than value. | Title | | 1 |

Longer value than title

| Title | | 1234567890 |

I’d like to get a similar behavior, or at least be able to selectively turn it on, for a header with joined columns. The above rules should still work if they take precedence requiring the greatest width, however as it stands I have to manually set widths evenly distributed across all the joined columns to create enough room for the joined header text. Not doing so results in the text getting cut off.

grid.prependHeaderRow().join(col1, col2).setText("Quite A Long Title");

| Quite a L…| | col1 | col2 | | 1| 2|

So in short, could the grid just widen grouped columns to accommodate the joined header text when it would be required?

See attached illustration.

___
 Imported from https://dev.vaadin.com/ issue #19239
